resources/lib/database/db_base_sqlite.py

```diff
@@ -39,26 +52,35 @@ def wrapper(*args, **kwargs):
             # If database is mysql pass to next decorator
             return func(*args, **kwargs)
         conn = None
+        global LIBSQLITE3_THREADSAFE
+        try:
+            LIBSQLITE3_THREADSAFE
+        except NameError:
+            LIBSQLITE3_THREADSAFE = is_libsqlite3_threadsafe()
         try:
             if not args[0].is_connected:
-                args[0].mutex.acquire()
-                args[0].conn = sql.connect(args[0].db_file_path,
-                                           isolation_level=CONN_ISOLATION_LEVEL)
+                if not LIBSQLITE3_THREADSAFE:
+                    args[0].mutex.acquire()
+                    args[0].conn = sql.connect(args[0].db_file_path,
+                                               isolation_level=CONN_ISOLATION_LEVEL,
+                                               check_same_thread = True)
+                else:
+                    args[0].conn = sql.connect(args[0].db_file_path,
+                                               isolation_level=CONN_ISOLATION_LEVEL,
+                                               check_same_thread = False)
```

Comment:
else not needed, just reuse same code line
```suggestion
                args[0].conn = sql.connect(args[0].db_file_path,
                                           isolation_level=CONN_ISOLATION_LEVEL,
                                           check_same_thread=not G.IS_SQLITE3_THREADSAFE)
```